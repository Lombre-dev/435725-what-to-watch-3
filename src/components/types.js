import PropTypes from 'prop-types';

export const Review = PropTypes.shape({
  id: PropTypes.number.isRequired,
  author: PropTypes.string.isRequired,
  score: PropTypes.number.isRequired,
  text: PropTypes.string.isRequired,
  date: PropTypes.number.isRequired,
});

export const Movie = PropTypes.shape({
  id: PropTypes.number.isRequired,
  title: PropTypes.string.isRequired,
  genres: PropTypes.arrayOf(PropTypes.string).isRequired,
  year: PropTypes.number.isRequired,
  poster: PropTypes.string,
  frames: PropTypes.arrayOf(PropTypes.string).isRequired,
  preview: PropTypes.string.isRequired,
  src: PropTypes.string.isRequired,
  ratingScore: PropTypes.number.isRequired,
  ratingReviewsCount: PropTypes.number.isRequired,
  description: PropTypes.string,
  story: PropTypes.string,
  duration: PropTypes.number.isRequired,
  director: PropTypes.string.isRequired,
  actors: PropTypes.arrayOf(PropTypes.string).isRequired,
  backgroundImage: PropTypes.string,
  backgroundColor: PropTypes.string,
  isFavorite: PropTypes.bool,
});
