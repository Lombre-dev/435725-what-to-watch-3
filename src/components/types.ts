import {AuthorizationStatus} from '../consts'

export type TReview = {
  id: number,
  author: string,
  score: number,
  text: string,
  date: number,
}

export type TMovie = {
  id: number,
  title: string,
  genres: string[],
  year: number,
  poster?: string,
  frames: string[],
  preview: string,
  src: string,
  ratingScore: number,
  ratingReviewsCount: number,
  description?: string,
  story?: string,
  duration: number,
  director: string,
  actors: string[],
  backgroundImage?: string,
  backgroundColor?: string,
  isFavorite?: boolean,
}

export type TUser = {
  id?: number,
  name?: string,
  email?: string,
  avatar?: string,
  authRequired?: boolean,
  authError?: number,
  authStatus?: AuthorizationStatus,
  favoriteMovies?: [],
}

export type TMatchParamsWithId = {
  params: {
    id: string;
  }
}
