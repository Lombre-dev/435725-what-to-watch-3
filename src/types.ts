export type TReview = {
  id: number;
  author: string;
  score: number;
  text: string;
  date: number;
}

export type TReviewData = {
  id: number;
  user: {
    id: number;
    name: string;
  };
  rating: number;
  comment: string;
  date: string;
};

export type TMovie = {
  id: number;
  title: string;
  genres: string[];
  year: number;
  poster?: string;
  frames: string[];
  preview: string;
  src: string;
  ratingScore: number;
  ratingReviewsCount: number;
  description?: string;
  story?: string;
  duration: number;
  director: string;
  actors: string[];
  backgroundImage?: string;
  backgroundColor?: string;
  isFavorite?: boolean;
}

export type TMovieData = {
  id: number;
  name: string;
  poster_image: string;
  preview_image: string;
  background_image: string;
  background_color: string;
  video_link: string;
  preview_video_link: string;
  description: string;
  rating: number;
  scores_count: number;
  director: string;
  starring: string[];
  run_time: number;
  genre: string;
  released: number;
  is_favorite: boolean;
};

export type TUser = {
  id?: number;
  name?: string;
  email?: string;
  avatar?: string;
  authRequired?: boolean;
  authError?: number;
  authStatus?: AuthorizationStatus;
  favoriteMovies?: [];
}

export type TUserData = {
  id: number;
  email: string;
  name: string;
  avatar_url: string;
}

export type TMatchParamsWithId = {
  params: {
    id: string;
  };
}

export enum AppPages {
  MAIN = `/`,
  LOGIN = `/login`,
  MY_LIST = `/mylist`,
  MOVIES = `/films`,
  PLAYER = `/player`,
}

export enum PlayerState {
  INITED = `inited`,
  LOADING = `loading`,
  PLAYING = `playing`,
  PAUSED = `paused`,
  ENDED = `ended`,
}

export enum ServerErrors {
  NOT_FOUND = 404,
  BAD_REQUEST = 400,
  UNAUTHORIZED = 401,
}

export enum AuthorizationErrorCode {
  INCORRECT_LOGIN = 1,
  AUTHORIZATION_FAIL = 2,
}

export enum AuthorizationStatus {
  NO_AUTH = `unauthorized`,
  AUTH = `authorized`,
}

export enum LoadingDataStatus {
  LOADING = `loading`,
  READY = `ready`,
  ERROR = `error`,
}
