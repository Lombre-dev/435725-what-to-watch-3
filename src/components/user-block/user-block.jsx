import React from 'react';
import {User} from '../types';

function UserBlock({user}) {
  return (
    <div className="user-block">
      {
        user && <div className="user-block__avatar">
          <img src={user.avatar} alt={user.name} width="63" height="63" />
        </div>
      }
      {
        !user && <a className="user-block__link" href="/sign-in">Sign in</a>
      }
    </div>
  );
}

UserBlock.propTypes = {
  user: User,
};

export default UserBlock;
