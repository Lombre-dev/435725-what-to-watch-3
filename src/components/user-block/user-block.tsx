import * as React from 'react';
import {connect} from 'react-redux';
import {Link} from 'react-router-dom';
import {AppPages, AuthorizationStatus} from '../../types';
import {getUserAuthStatus, getUserAvatar, getUserName} from '../../redux/user/selectors';

type TProps = {
  status?: AuthorizationStatus;
  name?: string;
  avatar?: string;
};

function UserBlock(props: TProps) {

  const {status, avatar, name} = props;

  return (
    <div className="user-block">
      {
        status === AuthorizationStatus.AUTH &&
        <Link to={AppPages.MY_LIST}>
          <div className="user-block__avatar">
            <img src={avatar} alt={name} width="63" height="63" />
          </div>
        </Link>
      }
      {
        status === AuthorizationStatus.NO_AUTH &&
        <Link className="user-block__link" to={AppPages.LOGIN}>Sign in</Link>
      }
    </div>
  );
}

function mapStateToProps(state) {
  return {
    status: getUserAuthStatus(state),
    name: getUserName(state),
    avatar: getUserAvatar(state),
  };
}

export {UserBlock};
export default connect(mapStateToProps)(UserBlock);
