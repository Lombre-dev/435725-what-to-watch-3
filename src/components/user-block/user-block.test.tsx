import * as React from 'react';
import {BrowserRouter} from 'react-router-dom';
import * as renderer from 'react-test-renderer';
import {AuthorizationStatus} from '../../types';
import {UserBlock} from './user-block';

const STATUS = AuthorizationStatus.AUTH;
const NAME = `Test`;
const AVATAR = `Avatar.jpg`;

describe(`<UserBlock />`, () => {

  it(`render should be match markup`, () => {

    const result = renderer
      .create(<BrowserRouter>
        <UserBlock
          status={STATUS}
          name={NAME}
          avatar={AVATAR}
        />
      </BrowserRouter>)
      .toJSON();

    expect(result).toMatchSnapshot();
  });
});
