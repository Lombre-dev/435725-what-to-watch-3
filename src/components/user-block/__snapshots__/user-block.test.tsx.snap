// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserBlock /> render should be match markup 1`] = `
<div
  className="user-block"
>
  <a
    href="/mylist"
    onClick={[Function]}
  >
    <div
      className="user-block__avatar"
    >
      <img
        alt="Test"
        height="63"
        src="Avatar.jpg"
        width="63"
      />
    </div>
  </a>
</div>
`;
